### DISCUSSION TITLE
Saving fragments in the right folders

### DISCUSSION THREAD

**Discussion by MishaPhysics** [ImageMagick/ImageMagick#6516] [Help] [answered]
### ImageMagick version

7.1.1-14

### Operating system, version and so on

Windows 10

### Description

Hi,
I'm wondering if it's possible to do the following thing using ImageMagick.
For example:
1) I have a picture "image.png" of 16384x16384 resolution in "C:\Users\xolms\Downloads\Folder".
2) I cut this file using "magick mogrify -crop 2048x2048 image.png" in cmd (or PowerShell), and obtain 64 pictures named from "image-0.png" to "image-63.png" .
3) I convert all these 64 png files into dds format using "magick mogrify -format dds *.png".
4) I would like to create 8 folders named from "002360" to "002367", then moving my 64 dds files into them and renaming as follows:
a) Files from "image-0.dds" to "image-7.dds" move to "C:\Users\xolms\Downloads\Folder\002360", and renaming them from "003760.png" to "003767.png".
b) Files from "image-8.dds" to "image-15.dds" move to "C:\Users\xolms\Downloads\Folder\002361", and renaming them from "003760.png" to "003767.png".
...
h) Files from "image-56.dds" to "image-63.dds" move to "C:\Users\xolms\Downloads\Folder\002360", and renaming them from "003760.png" to "003767.png".

Is it possible to make a script for doing this? I would like to specify the number of folders and dds files (they are the same), name of the first folder ("002360") and the name of the first dds file (003760).

Could you please advise something for this (or other more suitable forum for this question)? Also, I have Linux Mint, if it will be easier to use its terminal. Sorry if this question is inappropriate here.

**Comment by snibgo** [comment, marked as answer]
Step 4 could be scripted in the shell of your choice. The shell commands would need to do arithmetic using environment variables, with leading zeros, make directories, move/rename files, and loop through files.

If you don't know how to do those operations in the shell of your choice, I suggest you read the documentation.
